Failure to compile on master due to `error: unused variable 'str'` in Ragel `request_parser.rl` with Clang 15

I used the following options:

```sh
./configure.py --mode=release  --c-compiler=clang --compiler=clang++ \
	--verbose --disable-dpdk --enable-hwloc --enable-alloc-failure-injector \
	--enable-task-backtrace --enable-debug-shared-ptr --enable-io_uring \
	--allocator-page-size 4096 --without-tests --without-apps --without-demos \
	--split-dwarf --compile-commands-json --heap-profiling --prefix /usr/local
```

CMake configure:
---
```sh
[user@MSI seastar]$ ./configure.py --mode=release  --c-compiler=clang --compiler=clang++ --verbose --disable-dpdk --enable-hwloc --enable-alloc-failure-injector --enable-task-backtrace --enable-debug-shared-ptr --enable-io_uring --allocator-page-size 4096 --without-tests --without-apps --without-demos --split-dwarf --compile-commands-json --heap-profiling --prefix /usr/local
Running CMake in '/home/user/projects/bustub-cpp-reimpl/third_party/seastar/build/release' ...
cmake \
  -G \
  Ninja \
  ../.. \
  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
  -DCMAKE_C_COMPILER=clang \
  -DCMAKE_CXX_COMPILER=clang++ \
  -DCMAKE_CXX_STANDARD=20 \
  -DCMAKE_INSTALL_PREFIX=/usr/local \
  -DCMAKE_EXPORT_COMPILE_COMMANDS=yes \
  -DSeastar_API_LEVEL=6 \
  -DSeastar_SCHEDULING_GROUPS_COUNT=16 \
  -DSeastar_EXCLUDE_TESTS_FROM_ALL=yes \
  -DSeastar_EXCLUDE_APPS_FROM_ALL=yes \
  -DSeastar_EXCLUDE_DEMOS_FROM_ALL=yes \
  -DSeastar_CXX_FLAGS= \
  -DSeastar_LD_FLAGS= \
  -DSeastar_DPDK=no \
  -DSeastar_DPDK_MACHINE=native \
  -DSeastar_HWLOC=yes \
  -DSeastar_IO_URING=yes \
  -DSeastar_ALLOC_FAILURE_INJECTION=yes \
  -DSeastar_TASK_BACKTRACE=yes \
  -DSeastar_ALLOC_PAGE_SIZE=4096 \
  -DSeastar_SPLIT_DWARF=yes \
  -DSeastar_HEAP_PROFILING=yes \
  -DSeastar_DEFERRED_ACTION_REQUIRE_NOEXCEPT=yes \
  -DSeastar_UNUSED_RESULT_ERROR=no \
  -DSeastar_DEBUG_SHARED_PTR=yes
-- The CXX compiler identification is Clang 15.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Boost: /usr/include (found suitable version "1.78.0", minimum required is "1.73.0")  
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found Boost: /usr/include (found suitable version "1.78.0", minimum required is "1.73.0") found components: filesystem program_options thread unit_test_framework chrono atomic 
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0")
-- Checking for module 'libcares'
--   Found libcares, version 1.17.2
-- Found c-ares: /usr/lib64/libcares.so (found suitable version "1.17.2", minimum required is "1.13") 
-- Found cryptopp: /usr/lib64/libcryptopp.so (Required is at least version "5.6.5") 
-- Checking for one of the modules 'liblz4'
-- Found lz4: /usr/lib64/liblz4.so (found suitable version "1.9.3", minimum required is "1.7.3") 
-- Looking for LZ4_compress_default
-- Looking for LZ4_compress_default - found
-- Performing Test Concepts_FOUND
-- Performing Test Concepts_FOUND - Failed
-- Checking for one of the modules 'gnutls'
-- Found GnuTLS: /usr/lib64/libgnutls.so (found suitable version "3.7.8", minimum required is "3.3.26") 
-- Checking for one of the modules 'liburing'
-- Performing Test HAVE_IOURING_FEATURES
-- Performing Test HAVE_IOURING_FEATURES - Success
-- Found LibUring: /usr/lib64/liburing.so (Required is at least version "2.0")
-- Performing Test LinuxMembarrier_FOUND
-- Performing Test LinuxMembarrier_FOUND - Success
-- Performing Test Sanitizers_ADDRESS_FOUND
-- Performing Test Sanitizers_ADDRESS_FOUND - Success
-- Performing Test Sanitizers_UNDEFINED_BEHAVIOR_FOUND
-- Performing Test Sanitizers_UNDEFINED_BEHAVIOR_FOUND - Success
-- Performing Test Sanitizers_FIBER_SUPPORT
-- Performing Test Sanitizers_FIBER_SUPPORT - Success
-- Found Sanitizers: -fsanitize=address
-- Performing Test CxxSourceLocation_SUPPORTED
-- Performing Test CxxSourceLocation_SUPPORTED - Success
-- Performing Test StdAtomic_EXPLICIT_LINK
-- Performing Test StdAtomic_EXPLICIT_LINK - Success
-- Found StdAtomic: -latomic
-- Checking for one of the modules 'hwloc'
-- Found hwloc: /usr/lib64/libhwloc.so (found suitable version "2.5.0", minimum required is "1.11.2") 
-- Found lksctp-tools: /usr/lib64/libsctp.so  
-- Found numactl: /usr/lib64/libnuma.so  
-- Checking for one of the modules 'yaml-cpp'
-- Found yaml-cpp: /usr/lib64/libyaml-cpp.so (found suitable version "0.6.3", minimum required is "0.5.1") 
-- Found ragel: /usr/bin/ragel (found suitable version "7.0.0.12", minimum required is "6.10")
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - found
-- Found PthreadSetName: 1
-- Checking for one of the modules 'valgrind'
-- Found Valgrind: /usr/include  
-- Performing Test tmp_MaybeUninitialized_FOUND
-- Performing Test tmp_MaybeUninitialized_FOUND - Failed
-- Performing Test tmp_ErrorUnused_FOUND
-- Performing Test tmp_ErrorUnused_FOUND - Success
-- Performing Test tmp_ErrorWarnings_FOUND
-- Performing Test tmp_ErrorWarnings_FOUND - Success
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER


-- Build files have been written to: /home/user/projects/bustub-cpp-reimpl/third_party/seastar/build/release
```

Ninja build:
---
```cpp
[user@MSI seastar]$ ninja -C build/release
ninja: Entering directory `build/release`
[15/93] Building CXX object CMakeFiles/seastar.dir/src/core/prometheus.cc.o
FAILED: CMakeFiles/seastar.dir/src/core/prometheus.cc.o
/usr/bin/clang++ -DFMT_SHARED -DSEASTAR_API_LEVEL=6 -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFERRED_ACTION_REQUIRE_NOEXCEPT -DSEASTAR_ENABLE_ALLOC_FAILURE_INJECTION -DSEASTAR_HAS_MEMBARRIER -DSEASTAR_HAVE_ASAN_FIBER_SUPPORT -DSEASTAR_
HAVE_HWLOC -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_URING -DSEASTAR_OVERRIDE_ALLOCATOR_PAGE_SIZE=4096 -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_TASK_BACKTRACE -I/home/user/projects/bustub-cpp-reimp
l/third_party/seastar/include -I/home/user/projects/bustub-cpp-reimpl/third_party/seastar/build/release/gen/include -I/home/user/projects/bustub-cpp-reimpl/third_party/seastar/src -I/home/user/projects/bustub-cpp-reimpl/third_party/
seastar/build/release/gen/src -O2 -g -DNDEBUG -U_FORTIFY_SOURCE -DSEASTAR_SSTRING -gsplit-dwarf -Wno-error=unused-result "-Wno-error=#warnings" -fvisibility=hidden -UNDEBUG -Wall -Werror -Wno-array-bounds -Wno-error=deprecated-decla
rations -gz -std=gnu++20 -MD -MT CMakeFiles/seastar.dir/src/core/prometheus.cc.o -MF CMakeFiles/seastar.dir/src/core/prometheus.cc.o.d -o CMakeFiles/seastar.dir/src/core/prometheus.cc.o -c /home/user/projects/bustub-cpp-reimpl/third_party/seastar/src/core/prometheus.cc
In file included from /home/user/projects/bustub-cpp-reimpl/third_party/seastar/src/core/prometheus.cc:22:
In file included from /home/user/projects/bustub-cpp-reimpl/third_party/seastar/include/seastar/core/prometheus.hh:24:
In file included from /home/user/projects/bustub-cpp-reimpl/third_party/seastar/include/seastar/http/httpd.hh:24:
/home/user/projects/bustub-cpp-reimpl/third_party/seastar/src/http/request_parser.rl:55:9: error: unused variable 'str' [-Werror,-Wunused-variable]
                        auto str = [this, &g, &p] { g.mark_end(p); return get_str(); };
                             ^
1 error generated.
[32/93] Building CXX object CMakeFiles/seastar.dir/src/core/reactor.cc.o
ninja: build stopped: subcommand failed.
```

--- a/src/http/request_parser.rl
+++ b/src/http/request_parser.rl
@@ -150,7 +150,7 @@ public:
     }
     char* parse(char* p, char* pe, char* eof) {
         sstring_builder::guard g(_builder, p, pe);
-        auto str = [this, &g, &p] { g.mark_end(p); return get_str(); };
+        [[maybe_unused]] auto str = [this, &g, &p] { g.mark_end(p); return get_str(); };
         bool done = false;
         if (p != pe) {
             _state = state::error;

